libs/ledger/include/ledger/chaincode/smart_contract_manager.hpp

```diff
@@ -32,6 +33,7 @@ class SmartContractManager : public Contract
   static constexpr char const *LOGGING_NAME = "SmartContractManager";
 
   static storage::ResourceAddress CreateAddressForContract(Identifier const &contract_id);
+  static storage::ResourceAddress CreateAddressForSynergeticContract(Digest const &digest);
```

Comment:
With this construct, is it possible to access the same states in a `@action` and `@work` function? Ideally this should be possible.

Reply:
It is not currently possible. I had imagined that we would support this alongside the "contract to contract" work as this is a more natural place to support this.

In general, this function is just about looking up where the synergetic and smart contract code is stored in the state database. In this way they are treated as the same in that they are stored using hash.

Reply:
OK, noted. I'd suggest to just briefly discuss this in person.